mojo/stdlib/stdlib/collections/string/string.mojo

```diff
@@ -284,7 +284,14 @@ struct String(
 
     fn __init__[
         *Ts: Writable,
-    ](out self, *args: *Ts, sep: StaticString = "", end: StaticString = ""):
+        O1: ImmutableOrigin = StaticConstantOrigin,
+        O2: ImmutableOrigin = StaticConstantOrigin,
+    ](
+        out self,
+        *args: *Ts,
+        sep: StringSlice[O1] = rebind[StringSlice[O1]](StaticString()),
```

Comment:
This LGTM, but can we add a simple unit test that passes `sep` and `end` explicitly as `StringSlice`s to ensure there are no issues?  I understand the previous uses/tests with `StaticString` will work as before, but now that we've loosened the requirements in a sense, I want to ensure we have tests with the lowest requirement as well.